### PR TITLE
home work #10

### PR DESCRIPTION
**Pull request by undef1nd**

### CODE REVIEW

File: homework/taniab/tania_hw_10.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        path = os.path.join(self.target_folder, img_name)
+        with open(path, 'wb') as img_file:
+            for chunk in resp.iter_content(chunk_size=64):
+                img_file.write(chunk)
+        print(f'Successfully downloaded as {path}')
+
+    def download_all(self):
+        for url in self.urls_list:
+            response = requests.get(url)
+            if self._validate_response(response) and self._validate_size(response) and self._validate_type(response):
```
Comment (aodarc) [inline homework/taniab/tania_hw_10.py]: use `all()` function to avoid many `and`

Comment (aodarc) [inline homework/taniab/tania_hw_10.py]: I think also good idea to collect unsuccessfully downloaded images in separate variable

**Review by aodarc — CHANGES_REQUESTED** [review]

File: homework/taniab/tania_hw_10.py (hunk 1 above)
Comment (undef1nd) [reply homework/taniab/tania_hw_10.py]: @aodarc , fixed. Please, take a look.

Comment (undef1nd) [reply homework/taniab/tania_hw_10.py]: @aodarc, added as well. Please, approve PR.

File: homework/taniab/tania_hw_10.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    def _save(self, resp):
+        img_name = resp.url.split('/')[-1]
+        path = os.path.join(self.target_folder, img_name)
+        with open(path, 'wb') as img_file:
+            for chunk in resp.iter_content(chunk_size=64):
+                img_file.write(chunk)
+        print(f'Successfully downloaded as {path}')
+
+    def download_all(self):
+        failed_urls = []
```
Comment (aodarc) [inline homework/taniab/tania_hw_10.py]: I guess it might be instance attribute, but now it's not matter. for now)

**Review by aodarc — APPROVED** [review]

**Pull request merged**